### PR TITLE
TarResponseErrorMap: to kotlin

### PR DESCRIPTION
**Pull request by RingerJK**
## Description

Migrate Java to Kotlin `TarResponseErrorMap` and `OfflineRouteError` [#issue](https://github.com/mapbox/navigation-sdks/issues/167)

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMap.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+internal class TarResponseErrorMap(private val errorCodes: HashMap<Int, String>) {
+
+    companion object {
+        private const val TILES_ACCESS_TOKEN_ERROR_CODE = 402
+        private const val BOUNDING_BOX_ERROR_CODE = 422
+        private const val TILES_ACCESS_TOKEN_ERROR_MESSAGE =
+            "Unable to fetch tiles: Before you can fetch " + "routing tiles you must obtain an enterprise access token. Please contact us at support@mapbox.com"
+        private const val BOUNDING_BOX_ERROR_MESSAGE =
+            "Unable to fetch tiles: The bounding box you have " + "specified is too large. Please select a smaller box and try again."
+        private const val ERROR_MESSAGE_FORMAT = "Error code %s: %s"
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMap.kt]: > Property "ERROR_MESSAGE_FORMAT" is never used

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMap.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    init {
+        errorCodes[TILES_ACCESS_TOKEN_ERROR_CODE] = TILES_ACCESS_TOKEN_ERROR_MESSAGE
+        errorCodes[BOUNDING_BOX_ERROR_CODE] = BOUNDING_BOX_ERROR_MESSAGE
+    }
+
+    fun buildErrorMessageWith(response: Response<ResponseBody>): String {
+        var errorMessage = errorCodes[response.code()]
+        if (errorMessage == null) {
+            errorMessage = "Error code ${response.code()}: ${response.message()}"
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMap.kt]: Could we fix `ERROR_MESSAGE_FORMAT` removing ` %s: %s` and use it here instead?

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMap.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package com.mapbox.services.android.navigation.v5.navigation
+
+import java.util.HashMap
+import okhttp3.ResponseBody
+import retrofit2.Response
+
+internal class TarResponseErrorMap(private val errorCodes: HashMap<Int, String>) {
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMap.kt]: `TarResponseErrorMap` is only instantiated in `TarFetchedCallback` and it's passing a plain `errorCodes` `HashMap`, should we remove this parameter and create it at `init` time?

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMapTest.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import org.junit.Test
+import retrofit2.Response
+
+class TarResponseErrorMapTest {
+
+    @Test
+    fun buildErrorMessage_402messageIsCreated() {
+        val errorCodes = HashMap<Int, String>()
+        val response = mockk<Response<ResponseBody>>()
+
```
Comment (Guardiola31337) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMapTest.kt]: NIT no new line needed

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMapTest.kt [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        every { response.code() } returns 402
+
```
Comment (Guardiola31337) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMapTest.kt]: NIT no new line needed

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMapTest.kt [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        val errorMap = TarResponseErrorMap(errorCodes)
+        val errorMessage = errorMap.buildErrorMessageWith(response)
```
Comment (Guardiola31337) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMapTest.kt]: NIT new line
It's good to structure the tests following the triple A 👀 https://speakerdeck.com/guardiola31337/elegant-unit-testing-mobilization-2016?slide=40 http://wiki.c2.com/?ArrangeActAssert

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMapTest.kt [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+        assertTrue(errorMessage.contains("Please contact us at support@mapbox.com"))
+    }
+
+    @Test
+    fun buildErrorMessage_messageIsCreatedForCodeNotFound() {
+        val errorCodes = HashMap<Int, String>()
+        val response = mockk<Response<ResponseBody>>()
+
```
Comment (Guardiola31337) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMapTest.kt]: Same here re: triple A / new lines

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMap.kt (hunk 3 above)
Comment (RingerJK) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMap.kt]: need for tests, I've added default param `private val errorCodes: HashMap<Int, String> = HashMap()`

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMap.kt (hunk 2 above)
Comment (RingerJK) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMap.kt]: done

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMapTest.kt (hunk 4 above)
Comment (RingerJK) [reply libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMapTest.kt]: done

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMapTest.kt (hunk 5 above)
Comment (RingerJK) [reply libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMapTest.kt]: done

**Pull request merged**